modules/imap/functions.php

```diff
@@ -1425,3 +1425,50 @@ function parse_sieve_config_host($host) {
     // $host = '$scheme://'.$host;
     return [$host, $port, $tls];
 }}
+
+if (!hm_exists('connect_to_imap_server')) {
+    function connect_to_imap_server($address, $name, $port, $user, $pass, $tls, $imap_sieve_host, $enableSieve, $type, $context, $hidden = false) {
+        Hm_IMAP_List::init($context->user_config, $context->session);
+        $imap_list = array(
+            'name' => $name,
+            'server' => $address,
+            'hide' => false,
+            'port' => $port,
+            'user' => $user,
+            'pass' => $pass,
+            'tls' => $tls);
+
+        if ($type === 'jmap') {
+            $imap_list['type'] = 'jmap';
+            $imap_list['hide'] = $hidden;
+            $imap_list['port'] = false;
+            $imap_list['tls'] = false;
+        }
+
+        if (isset($imap_sieve_host) && $imap_sieve_host) {
+            $imap_list['sieve_config_host'] = $imap_sieve_host;
+        }
+
+        Hm_IMAP_List::add($imap_list);
```

Comment:
Add method returns the ID of the added server. There is no need to dump the whole list, get the keys and pop the last key to get the server id.